bundles/framework/divmanazer/component/FileInput.js

```diff
@@ -80,7 +112,90 @@ Oskari.clazz.define('Oskari.userinterface.component.FileInput', function () {
                 }
             });
             this.setElement( form.parent() );
-            return this.getElement();
+            */
+        },
+        bindBasicUpload: function() {
+            var me = this;
+            var elem = this.getElement();
+            var input = elem.find('input[type="file"]');
+            input.change(function(e){
+                me.handleFileList(e.target.files);
+            });
+        },
+        handleFileList: function (fileList){
+            var me = this;
+            var opts = this.options;
+            var files = fileList;
+            var file;
+
+            if (opts.allowMultipleFiles === true){
+                Object.keys( files ).forEach( function ( key ) {
+                    file = files[key];
+                    if (me.validateFile(file) === true){
+                        me.files.push(file);
+                    } else {
+                        //TODO or keep valid files
+                        me.files = [];
+                        return;
+                    }
+                });
+            } else {
+                if (files.length > 1){
+                    this.showPopup(this.loc('fileInput.error'),this.loc('fileInput.multipleNotAllowed'));
+                } else {
+                    file = files[0];
+                    if (file && this.validateFile(file) === true){
+                        this.files = [file];
+                    } else {
+                        this.files = [];
+                    }
+                }
+            }
+            if (this.isAdvancedUpload){
+                this.updateFileList();
+            }
+        },
+        updateFileList: function (){
+            var fileNameElem = this.getElement().find('.box__uploaded');
+            var fileNames;
+            var files = this.files;
+            if (files.length === 0){
+                fileNameElem.text("");
+            } else {
+                fileNames = files[0].name;
+                for (var i = 1; i < files.length; i++){
+                    fileNames += ", " + files[i].name;
+                }
```

Comment:
You could also do this with:

	var fileNames = files.map(function (file) {
		return file.name;
	}).join(', ');

Reply:
I guess you wouldn't even need the if-statement if you did it like this.